### PR TITLE
refactor: rename `InnerServer` to `Storage`

### PR DESCRIPTION
**Pull request by clark1013**
Related to: https://github.com/pingcap-incubator/tinykv/issues/109

I have renamed below:
```
inner_server -> storage
InnerServer -> Storage
innerServer -> storage
mem_server -> mem_storage
standalone_server -> standalone_storage
raft_server -> raft_storage
DBReader -> StorageReader
is(inner server) -> s(storage)
ris(raft inner server) -> rs(raft storage)
```
But I am still not sure if `tinykv/kv/storage/raft_storage/raft_server.go` & `raft_serverpb` should be renamed too.

### CODE REVIEW

File: kv/transaction/doc.go [hunk 1]
```diff
@@ -1,8 +1,8 @@
 package transaction
 
 // The transaction package implements TinyKV's 'transaction' layer. This takes incoming requests from the tikv/server.go as
-// input and turns them into reads and writes of the underlying key/value store (defined by InnerServer in tikv/server.go).
-// The InnerServer handles communicating with other nodes and writing data to disk. The transaction layer must
+// input and turns them into reads and writes of the underlying key/value store (defined by Storage in tikv/server.go).
+// The Storage handles communicating with other nodes and writing data to disk. The transaction layer must
```
Comment (nrc) [inline kv/transaction/doc.go]:
```suggestion
// The storage engine handles communicating with other nodes and writing data to disk. The transaction layer must
```

File: kv/transaction/doc.go [hunk 2]
```diff
@@ -13,13 +13,13 @@ package transaction
 // is executed atomically.
 //
 // *Locks* are used to implement TinySQL transactions. Setting or checking a lock in a TinySQL transaction is lowered to
-// writing or reading a key and value in the InnerServer store.
+// writing or reading a key and value in the Storage store.
```
Comment (nrc) [inline kv/transaction/doc.go]:
```suggestion
// writing or reading a key and value in the store.
```

**Comment by clark1013** [comment]
Thank you for your advice, it helps a lot, `storage engine` is truly the words of a good.

**Comment by clark1013** [comment]
> Thanks! LGTM overall, seems forget to rename `raft_server.go` to `raft_storage.go`

I'm a little puzzled if I should do that. Since raft_storage package have `raft_server.go` & `raft_client.go` seems to be a good choice.
I'm also not sure if I should rename `raft_serverpb` to `raft_storagepb` too.
Suggestion is need here.

**Comment by clark1013** [comment]
> Would you mind updating the names in course material too? https://docs.google.com/document/d/1Z3OrUOq28a44UaipQiagwfM_py6gS9nLz-znk6jtMpU/edit#

Sure, I'm glad to. May be tonight.

**Comment by Connor1996** [comment]
> > Thanks! LGTM overall, seems forget to rename `raft_server.go` to `raft_storage.go`
> 
> I'm a little puzzled if I should do that. Since raft_storage package have `raft_server.go` & `raft_client.go` seems to be a good choice.
> I'm also not sure if I should rename `raft_serverpb` to `raft_storagepb` too.
> Suggestion is need here.

reasonable, LGTM

**Pull request merged**